terraform_files/oci/04_load_balancer.tf

```diff
@@ -0,0 +1,176 @@
+    ],
+    var.oci_extra_lb_nsg_oicds # e.g.: allow access to ci-machine to reach the API endpoint
+  )
+}
+
+locals {
+  cluster_nlb_public_ip  = one([for ip in oci_network_load_balancer_network_load_balancer.nlb.ip_addresses : ip.ip_address if ip.is_public])
+  cluster_nlb_private_ip = one([for ip in oci_network_load_balancer_network_load_balancer.nlb.ip_addresses : ip.ip_address if !ip.is_public])
+}
+
+# Backendset definitions
+
+resource "oci_network_load_balancer_network_load_balancers_backend_sets_unified" "nlb-bes-api" {
+  name                     = "bes-api"
+  network_load_balancer_id = oci_network_load_balancer_network_load_balancer.nlb.id
+  policy                   = "FIVE_TUPLE"
+  is_preserve_source       = false
+
+  health_checker {
+    port        = 6443
+    protocol    = "HTTPS"
+    url_path    = "/readyz"
+    return_code = 200
+  }
+
+  dynamic "backends" {
+    for_each = oci_core_instance.master
+    content {
+      port      = 6443
+      target_id = backends.value["id"]
+    }
+  }
+
+  depends_on = [
+    oci_network_load_balancer_network_load_balancer.nlb
+  ]
+}
+
+resource "oci_network_load_balancer_network_load_balancers_backend_sets_unified" "nlb-bes-mcs" {
+  name                     = "bes-mcs"
+  network_load_balancer_id = oci_network_load_balancer_network_load_balancer.nlb.id
+  policy                   = "FIVE_TUPLE"
+  is_preserve_source       = false
+
+  health_checker {
+    port        = 22623
+    protocol    = "HTTPS"
+    url_path    = "/healthz"
+    return_code = 200
+  }
+
+  dynamic "backends" {
+    for_each = oci_core_instance.master
+    content {
+      port      = 22623
+      target_id = backends.value["id"]
+    }
+  }
+
+  depends_on = [
+    oci_network_load_balancer_network_load_balancers_backend_sets_unified.nlb-bes-api
+  ]
+}
+
+resource "oci_network_load_balancer_network_load_balancers_backend_sets_unified" "nlb-bes-https" {
+  name                     = "bes-https"
+  network_load_balancer_id = oci_network_load_balancer_network_load_balancer.nlb.id
+  policy                   = "FIVE_TUPLE"
+  is_preserve_source       = false
+
+  health_checker {
+    port     = 443
+    protocol = "TCP"
+  }
+
+  dynamic "backends" {
+    for_each = oci_core_instance.worker
+    content {
+      port      = 443
+      target_id = backends.value["id"]
+    }
+  }
+
+  depends_on = [
+    oci_network_load_balancer_network_load_balancers_backend_sets_unified.nlb-bes-mcs
+  ]
+}
+
+resource "oci_network_load_balancer_network_load_balancers_backend_sets_unified" "nlb-bes-http" {
+  name                     = "bes-http"
+  network_load_balancer_id = oci_network_load_balancer_network_load_balancer.nlb.id
+  policy                   = "FIVE_TUPLE"
+  is_preserve_source       = false
+
+  health_checker {
+    port     = 80
+    protocol = "TCP"
+  }
+
+  dynamic "backends" {
+    for_each = oci_core_instance.worker
```

Comment:
```suggestion
    # for SNO and compact clusters, attach the masters to the HTTP backend
    for_each = if var.workers_count > 0 ? oci_core_instance.worker : oci_core_instance.master
```

Reply:
You meant without the if, right?

Reply:
I wrote it from my mind, so maybe there's not if in the ternary operator :)